File: .sdk/tm/py/utility/make_error.py
# RemoteJobsApiRssFeed SDK utility: make_error

from __future__ import annotations
from core.operation import RemoteJobsApiRssFeedOperation
from core.result import RemoteJobsApiRssFeedResult
from core.control import RemoteJobsApiRssFeedControl
from core.error import RemoteJobsApiRssFeedError


def make_error_util(ctx, err):
    if ctx is None:
        from core.context import RemoteJobsApiRssFeedContext
        ctx = RemoteJobsApiRssFeedContext({}, None)

    op = ctx.op
    if op is None:
        op = RemoteJobsApiRssFeedOperation({})
    opname = op.name
    if opname == "" or opname == "_":
        opname = "unknown operation"

    result = ctx.result
    if result is None:
        result = RemoteJobsApiRssFeedResult({})
    result.ok = False

    if err is None:
        err = result.err
    if err is None:
        err = ctx.make_error("unknown", "unknown error")

    errmsg = ""
    if isinstance(err, RemoteJobsApiRssFeedError):
        errmsg = err.msg
    elif hasattr(err, "msg") and err.msg is not None:
        errmsg = err.msg
    elif isinstance(err, str):
        errmsg = err
    else:
        errmsg = str(err)

    msg = "RemoteJobsApiRssFeedSDK: " + opname + ": " + errmsg
    msg = ctx.utility.clean(ctx, msg)

    result.err = None

    spec = ctx.spec

    if ctx.ctrl.explain is not None:
        ctx.ctrl.explain["err"] = {"message": msg}

    sdk_err = RemoteJobsApiRssFeedError("", msg, ctx)
    sdk_err.result = ctx.utility.clean(ctx, result)
    sdk_err.spec = ctx.utility.clean(ctx, spec)

    if isinstance(err, RemoteJobsApiRssFeedError):
        sdk_err.code = err.code

    ctx.ctrl.err = sdk_err

    if ctx.ctrl.throw_err is False:
        return result.resdata

    raise sdk_err

File: rb/utility/make_error.rb
# RemoteJobsApiRssFeed SDK utility: make_error
require_relative '../core/operation'
require_relative '../core/result'
require_relative '../core/error'
module RemoteJobsApiRssFeedUtilities
  MakeError = ->(ctx, err) {
    if ctx.nil?
      require_relative '../core/context'
      ctx = RemoteJobsApiRssFeedContext.new({}, nil)
    end
    op = ctx.op || RemoteJobsApiRssFeedOperation.new({})
    opname = op.name
    opname = "unknown operation" if opname.empty? || opname == "_"

    result = ctx.result || RemoteJobsApiRssFeedResult.new({})
    result.ok = false

    err = result.err if err.nil?
    err = ctx.make_error("unknown", "unknown error") if err.nil?

    errmsg = err.is_a?(RemoteJobsApiRssFeedError) ? err.msg : err.to_s
    msg = "RemoteJobsApiRssFeedSDK: #{opname}: #{errmsg}"
    msg = ctx.utility.clean.call(ctx, msg)

    result.err = nil
    spec = ctx.spec

    if ctx.ctrl.explain
      ctx.ctrl.explain["err"] = { "message" => msg }
    end

    sdk_err = RemoteJobsApiRssFeedError.new("", msg, ctx)
    sdk_err.result = ctx.utility.clean.call(ctx, result)
    sdk_err.spec = ctx.utility.clean.call(ctx, spec)
    sdk_err.code = err.code if err.is_a?(RemoteJobsApiRssFeedError)

    ctx.ctrl.err = sdk_err

    # Opt-out escape hatch: when throwing is explicitly disabled, return the
    # bare result data instead of raising.
    if ctx.ctrl.throw_err == false
      return result.resdata
    end
    # Default idiomatic path: raise the already-constructed exception.
    raise sdk_err
  }
end
